docs/content/sips/draft-000-templates.md

```diff
@@ -0,0 +1,365 @@
+but complete application, including the application metadata and a starter
+component (code and wiring).
+
+It would be good to provide a way to add a component to an existing application.
+This is a little tricky because (a) it requires modification of an existing
+file, not just copying a file; and (b) the component templates have to be
+constrained by the trigger type of the application.
+
+@michelleN has also proposed that it should be possible to create _only_ the
+application metadata, up to and including the trigger type, but with no
+components.  (The `spin up` command would then be modified to detect if
+an app had not components and guide the user on how to add them.)  The
+user experience for this is a bit tricky.  One option is to have separate
+`spin init/spin new` verbs, but for some readers that will create ambiguity -
+I want to create an application, which one is right for me?  Another is
+to have flags on `spin new` to indicate "bare starter," "full starter,"
+or "incremental add," but this creates ceremony and cognitive load. A
+possible approach is to make it interactive - if no manifest exists,
+ask whether to create a bare one or one with a starter component.  But
+this may be a confusing question for users' first encounter with Spin!
+"I don't know, I just want to work!"
+
+_TODO: think about how bare and incremental `new` might impact the
+authoring of templates as well as the consumer experience._
+
+### Running a template without installing
+
+This could be something like:
+
+```
+spin new https://github.com/fermyon/spin/templates/http/rust
+```
+
+There would be no language option.
+
+We'd need to enforce that IDs couldn't look like URLs.
+
+### What's this 'default language' thing?
+
+A somewhat unformed idea that users who primarily use one language could set an
+environment variable or config file setting and have the template system default
+to using that.
+
+## User experience: installing templates
+
+The current install experience has two paths, remote and local.  The remote path
+takes a Git repository URL and branch name, and clones the entire branch into
+the local cache.  The local one copies a file tree into cache.  In both cases
+you need to supply a repository name.
+
+It would be good, if we can, to avoid doing a full git clone.  A repository
+can be quite large, and we are really interested only in a snapshot of the
+template files on the chosen branch.  We can avoid cloning history via the
+`--depth 1` option but there doesn't seem to be a great option for only
+copying the template files.
+
+```
+spin templates install --dir <PATH>
+spin templates install --git <URL> [--branch foo]
+```
+
+The current convention for Git sources is that the templates should be located
+at `/templates`.  I'm not sure whether directory sources expect a `templates`
+subdirectory or if you specify the full path - I thought the former but it
+seemed to work when I used the latter. But maybe I was wrong.
```

Comment:
The current convention (which has not been thought out carefully) is to expect a `templates` directory _in the root of the repository_.